### PR TITLE
Add the ability to create Docker images from running containers

### PR DESCRIPTION
**Pull request by viren-nadkarni**
### Summary

Updates the Docker wrapper to allow:
- Creating images from containers
- Removing images

### Tests

Includes integration tests.

### CODE REVIEW

File: localstack/utils/docker_utils.py [hunk 1]
```diff
@@ -1508,6 +1569,26 @@ def has_docker(self) -> bool:
...
+    def commit(
+        self,
+        container_name_or_id: str,
+        image_name: str,
+        image_tag: str,
+    ):
+        LOG.debug(
+            "Creating image from container %s as %s:%s", container_name_or_id, image_name, image_tag
+        )
+        container = self.client().containers.get(container_name_or_id)
```
Comment (dfangl) [inline localstack/utils/docker_utils.py:1589]: Please add exception handling here if the target container does not exist, including a test for the behavior.
Also wrapping the whole thing to map API errors to container exceptions would be nice.

File: tests/integration/docker_utils/conftest.py [hunk 2]
```diff
@@ -12,7 +12,9 @@ def _check_skip():
         pytest.skip("Docker is not available")
 
 
-@pytest.fixture(params=[CmdDockerClient(), SdkDockerClient()])
+@pytest.fixture(
+    params=[CmdDockerClient(), SdkDockerClient()], ids=["CmdDockerClient", "SdkDockerClient"]
```
Comment (dfangl) [inline tests/integration/docker_utils/conftest.py:16]: Nice touch adding the IDs!

**Review by dfangl — CHANGES_REQUESTED** [review]
Please add the exception handling for the commit method in the sdkclient, and a test for the case of non existent containers!
Otherwise, LGTM!

File: localstack/utils/docker_utils.py [hunk 3]
```diff
@@ -1508,6 +1569,31 @@ def has_docker(self) -> bool:
...
+        LOG.debug(
+            "Creating image from container %s as %s:%s", container_name_or_id, image_name, image_tag
+        )
+        try:
+            container = self.client().containers.get(container_name_or_id)
+        except NotFound:
+            raise NoSuchContainer(container_name_or_id)
+        except APIError:
+            raise ContainerException()
+        container.commit(repository=image_name, tag=image_tag)
```
Comment (dfangl) [inline localstack/utils/docker_utils.py:1595]: Could you move the commit line also into the try block, so APIErrors from this call are mapped to ContainerException as well?

**Review by dfangl — CHANGES_REQUESTED** [review]
One more nit, then I am happy :)

File: localstack/utils/docker_utils.py (hunk 3 above)
Comment (viren-nadkarni) [reply localstack/utils/docker_utils.py]: Good catch. Fixed in 118e1ca8

**Review by dfangl — APPROVED** [review]
LGTM! Nice changes!

**Pull request merged**